### PR TITLE
feat: implement Nuxt module

### PR DESCRIPTION
**Pull request by markbrockhoff**
Relates to #1247

This PR adds a basic nuxt module to integrate onyx into nuxt projects. It just adds the global styles and auto imports all onyx components, more features may be added later.

## Checklist

- [x] The added / edited code has been documented with [JSDoc](https://jsdoc.app/about-getting-started)
- [x] All changes are documented in the documentation app (folder `apps/docs`)
- [x] If a new component is added, at least one [Playwright screenshot test](https://github.com/SchwarzIT/onyx/actions/workflows/playwright-screenshots.yml) is added
- [x] A changeset is added with `npx changeset add` if your changes should be released as npm package (because they affect the library usage)

### CODE REVIEW

**Comment by markbrockhoff** [comment]
I hope this works with your current release flow, I tried to adapt to the other packages. But please double check that e.g. all required scripts are in place and so on :)

File: apps/docs/src/development/index.md [hunk 1]
```diff
@@ -53,3 +53,19 @@ import { OnyxButton } from "sit-onyx";
   <OnyxButton label="Click me" />
 </template>
 ```
+
+## Nuxt
+
+To setup Onyx in a Nuxt Js project you can simply install the pre-configured Nuxt module for it:
```
Comment (BoppLi) [inline apps/docs/src/development/index.md:59]:
```suggestion
To setup Onyx in a [Nuxt.js](https://nuxt.com/) project you can simply install the pre-configured Nuxt module for it:
```
or
```suggestion
To setup Onyx in a [Nuxt JS](https://nuxt.com/) project you can simply install the pre-configured Nuxt module for it:
```
? :)

File: packages/nuxt/README.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Onyx Nuxt
```
Comment (BoppLi) [inline packages/nuxt/README.md:1]: please place the description in the docs app, so it appears here in the docs
image
like http://localhost:5173/development/packages/figma-utils.html

copy the content of this file into 
image

the readme of the actual packages are pointing to the docs with a logo on top, compare to
 `packages/figma-utils/README.md`

File: packages/nuxt/src/module.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import { addComponent, defineNuxtModule } from "@nuxt/kit";
+import * as onyx from "sit-onyx";
```
Comment (larsrickert) [inline packages/nuxt/src/module.ts:2]: Do you know if this is performant and stree-shakeable for bundle size? Or would we need a dynamic import for this?

File: packages/nuxt/src/module.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+export interface ModuleOptions {}
```
Comment (larsrickert) [inline packages/nuxt/src/module.ts:4]: There are `OnyxPluginOptions` that we could use here and also call the `createOnyx()` plugin.

But we can also do this in a follow up PR

File: packages/nuxt/test/fixtures/basic/app.vue [hunk 5]
```diff
@@ -0,0 +1,3 @@
+<template>
+  <OnyxHeadline is="h1">Hello Onyx</OnyxHeadline>
```
Comment (larsrickert) [inline packages/nuxt/test/fixtures/basic/app.vue:2]:
```suggestion
  <OnyxHeadline is="h1">Hello onyx</OnyxHeadline>
```

onyx is written in lowercase :D

**Review by larsrickert — CHANGES_REQUESTED** [review]
Thanks a lot for your contribution!

File: packages/nuxt/src/module.ts (hunk 3 above)
Comment (markbrockhoff) [reply packages/nuxt/src/module.ts:2]: Everything you see in this file will only run at build time, that's how nuxt modules work. It only registers the modules so they can be auto imported by nuxt. The import here is only used to get the names of the components for that. If one of the components is used in e.g. a page Nuxt will add `import { OnyxComponentName } from "sit-onyx"` to the page at build time. So all of this has nothing to do with the final bundle or the runtime.

File: packages/nuxt/src/module.ts (hunk 4 above)
Comment (markbrockhoff) [reply packages/nuxt/src/module.ts:4]: That's something to do once i18n and so on is integrated into this module. Like written in the description, this PR is just to bring the very basics of onyx into nuxt projects.

**Review by larsrickert — APPROVED** [review]
Great work!

**Pull request merged**